R/Rules-methods.R

```diff
@@ -305,28 +305,25 @@ setMethod("nextBest",
   definition = function(nextBest, doselimit, samples, model, data, ...) {
     doselimit <- ifelse(missing(doselimit) || length(doselimit) == 0, Inf, doselimit)
 
-    is_unacceptable_specified <- any(nextBest@unacceptable != c(1, 1))
-
     # Matrix with samples from the dose-tox curve at the dose grid points.
-    samples_prob <- sapply(data@doseGrid, prob, model = model, samples = samples)
+    prob_samples <- sapply(data@doseGrid, prob, model = model, samples = samples)
+    # Now compute probabilities to be in target and overdose tox interval.
+    prob_underdosing <- colMeans(prob_samples < nextBest@target[1])
+    prob_target <- colMeans(h_in_range(prob_samples, nextBest@target))
+    prob_overdose <- colMeans(h_in_range(prob_samples, nextBest@overdose, c(FALSE, TRUE)))
+    prob_mean <- colMeans(prob_samples)
+    prob_sd <- apply(prob_samples, 2, sd)
 
-    prob_underdosing <- colMeans(samples_prob < nextBest@target[1])
-    prob_target <- colMeans(h_in_range(samples_prob, nextBest@target))
-    prob_overdose <- colMeans(h_in_range(samples_prob, nextBest@overdose, c(FALSE, TRUE)))
-    prob_mean <- colMeans(samples_prob)
-    prob_sd <- apply(samples_prob, 2, sd)
+    is_unacceptable_specified <- any(nextBest@unacceptable != c(1, 1))
```

Comment:
so this is a fix for the `NextBestNCRMLoss` method right?

Reply:
Yap, it is a fix (that was applied in the Burak's branch and merged into main. so I just merged an updated main into this #273 branch) + few cosmetic changes after the fix, which I done in this branch.

Reply:
ah ok good. thx